src/http/client.cc

```diff
@@ -340,6 +340,22 @@ future<> client::make_request(request req, reply_handler handle, reply::status_t
     return do_with(std::move(req), std::move(handle), [this, expected] (request& req, reply_handler& handle) mutable {
         return with_connection([this, &req, &handle, expected] (connection& con) {
             return do_make_request(con, req, handle, expected);
+        }).handle_exception_type([this, &req, &handle, expected] (const transport_error& ex) {
+            if (ex.requests_submitted == 1) {
+                // It wasn't connection from pool, no need in making new one again
+                return make_exception_future<>(ex.cause);
+            }
+            // This creates one more connection without checking the limit, but
+            // since the old connection is closing it compensates the increase.
+            //
+            // However, in case new make_request() manages to creep in between
+            // it will temporarily create one extra connection, but it will not
+            // be held in pool anyway. Still, this is unlikely, _nr_connections
+            // decrease will happen in a waiting task and this place is chained
+            // directly after the put_connection()
```

Comment:
I still don't understand what this comment is saying. I think I just didn't understand how your pool works.